config/initializers/cors.rb

```diff
@@ -1,24 +1,31 @@
 # frozen_string_literal: true
 
 # Be sure to restart your server when you modify this file.
+# Read more: https://github.com/cyu/rack-cors
 
-origins_array = ENV['ALLOWED_ORIGINS']&.split(',')&.map(&:strip) || []
-# Handle Cross-Origin Resource Sharing (CORS) in order to accept cross-origin Ajax requests.
+require Rails.root.join('lib/origin_parser')
 
 Rails.application.config.middleware.insert_before 0, Rack::Cors do
   allow do
-    origins origins_array
-    resource '*', headers: :any, methods: %i[get post patch put delete], expose: ['Link']
+    # localhost and test domain origins
+    if Rails.env.development?
+      origins(%r{https?://localhost([:0-9]*)$})
+    elsif Rails.env.test?
+      origins(%r{https?://localhost([:0-9]*)$}, %r{https?://www\.example\.com$})
```

Comment:
apologies if I'm not getting something here, do we want example.com in here?

Reply:
Good question! It was something I initially copied over from the reference badge-service implementation (I assume it's used in a few specs there) but don't think it's strictly needed here at all...

I'll drop it as it's not really adding anything to keep it around.